lib/sisimai/bite/email/kddi.rb

```diff
@@ -0,0 +1,166 @@
+            )
+        /x,
+        :rfc822 => %r|\AContent-Type: message/rfc822\z|,
+        :error  => %r/Could not be delivered to:? /,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReFailure = {
+        mailboxfull: %r/As[ ]their[ ]mailbox[ ]is[ ]full/x,
+        norelaying:  %r/Due[ ]to[ ]the[ ]following[ ]SMTP[ ]relay[ ]error/x,
+        hostunknown: %r/As[ ]the[ ]remote[ ]domain[ ]doesnt[ ]exist/x,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'au by KDDI: http://www.au.kddi.com'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return []; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from au by KDDI
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['from']    =~ Re0[:from]
+        match += 1 if mhead['reply-to'] && mhead['reply-to'] =~ Re0[:'reply-to']
+        match += 1 if mhead['received'].find { |a| a =~ Re0[:received] }
+        return nil if match.zero?
+
+        require 'sisimai/string'
+        require 'sisimai/address'
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:'message-rfc822']).zero?` instead of `readcursor & Indicators[:'message-rfc822'] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->